api/service/store.go

```diff
@@ -35,23 +37,41 @@ func addEndpoint(apiContext *types.APIContext, schema *types.Schema, data map[st
 		return data, nil
 	}
 
+	spaceID, _ := data["spaceId"].(string)
+	space := ""
+	parts := strings.SplitN(spaceID, "-", 2)
```

Comment:
This will fail if spacename has a `-` in it.  `foo-bar-23452`

Reply:
How.....shouldn't spaceId always follow `s-randomstring` pattern cause you generate those name?